src/DelayBucket.cc

```diff
@@ -25,9 +27,19 @@ DelayBucket::stats(StoreEntry *entry)const
 void
 DelayBucket::update(DelaySpec const &rate, int incr)
 {
-    if (rate.restore_bps != -1 &&
-            (level() += rate.restore_bps * incr) > rate.max_bytes)
-        level() = rate.max_bytes;
+    if (rate.restore_bps == -1)
```

Comment:
IntegralProduct() assumes that we treat _all_ negative values specially, but this only checks for -1. The code directly below assumes that IntegralProduct() returns an undefined value only on overflows, but that is not how IntegralProduct() works. Should we check for all negative values here?
```suggestion
    // make sure an undefined IntegralProduct() result (below) implies overflow
    if (rate.restore_bps < 0) 
```